### PR TITLE
Bazelify ghc-lib

### PR DESCRIPTION
**Pull request by aherrmann-da**
### Pull Request Checklist

- [x] Read and understand the [contribution guidelines](https://github.com/digital-asset/daml/blob/main/CONTRIBUTING.md)
- [x] Include appropriate tests
- [ ] Set a descriptive title and thorough description
- [ ] Add a reference to the [issue this PR will solve](https://github.com/digital-asset/daml/issues), if appropriate
- [ ] Include changelog additions in one or more commit message bodies between the `CHANGELOG_BEGIN` and `CHANGELOG_END` tags
- [ ] Normal production system change, include purpose of change in description
- [ ] If you mean to change the status of a component, please make sure you keep [the Component Status page](https://github.com/digital-asset/daml/blob/main/docs/source/support/component-statuses.rst) up to date.

NOTE: CI is not automatically run on non-members pull-requests for security
reasons. The reviewer will have to comment with `/AzurePipelines run` to
trigger the build.

**TODO**
- [x] Remove `stack` from the `ghc-lib(-parser)` sdist generation rule inputs, on both Windows and Unix.
- [x] Factor out `ghc-lib` external dependencies from `WORKSPACE` into `bazel_tools/ghc-lib/repositories.bzl`.
- [x] Document the new rules in `bundle.bzl` and `dev_env_tool.bzl`.
- [x] Document the definitions in `bazel_tools/ghc-lib/defs.bzl|repositories.bzl`. In particular, document the indirection w.r.t. `BUILD.ghc(-lib)` to avoid refetch on changes.
- [x] Document in `bazel_tools/ghc-lib/README.md` the update and patching procedures. I.e. how to update the GHC revision, how to update the ghc-lib revision, how to update any of the needed patches.
- [x] Document how to build the various components. I.e. document how to build ghc-lib(-parser) sdists, and how to build the ghc-lib(-parser) Cabal libraries.
- [x] Add `cabal-update` runnable targets to `bazel_tools/ghc-lib/ghc-lib(-parser)` to update the respective Cabal files after the GHC revision changed.
- [x] Document an interactive development workflow based on a GHC checkout and Bazel's `--override_repository` flag.
- [x] Remove the old ghc-lib import
- [x] Change `GHC_REV` once https://github.com/digital-asset/ghc/pull/119 is merged

**Follow Up**
- [ ] Remove the old ghc-lib CI pipeline in `ci/da-ghc-lib/compile.yml`.
- [ ] Upstream `bundle.bzl` and `dev_env_tool.bzl` to `rules_sh`.

### CODE REVIEW

File: bazel_tools/sh/sh.bzl [hunk 1]
```diff
@@ -38,9 +39,8 @@ _sh_inline_script = rule(
         "output": attr.output(
             mandatory = True,
         ),
-        "_template": attr.label(
```
Comment (aherrmann-da) [inline bazel_tools/sh/sh.bzl:41]: Maybe add a comment why this is not a private attribute?

Comment (cbley-da) [reply bazel_tools/sh/sh.bzl]: It is a private attribute again.

File: build.ps1 [hunk 2]
```diff
@@ -8,6 +8,8 @@ Set-StrictMode -Version latest
 .\dev-env\windows\bin\dadew.ps1 sync
 .\dev-env\windows\bin\dadew.ps1 enable
 
+Get-ChildItem -Path C:\Users\u\dadew\scoop\apps\msys2\current\usr\bin
+
```
Comment (cbley-da) [inline build.ps1:12]: Should we remove this?

Comment (aherrmann-da) [reply build.ps1:12]: Good catch, yes, I think so.

**Review by garyverhaegen-da — COMMENTED** [review]
Looks good to me, but I'd prefer if someone from Language approved.

**Comment by cbley-da** [comment]
I have removed the patches to da-ghc as requested, rebased onto main, and the CI is green again. PTAL

Also see https://github.com/digital-asset/ghc/pull/119 which should be merged first.

**Review by sofiafaro-da — APPROVED** [review]
This looks great, thank you Claudio!

**Pull request merged**